pkg/services/sqlstore/migrations/ualert/silences.go

```diff
@@ -55,6 +56,46 @@ func (m *migration) addSilence(da dashAlert, rule *alertRule) error {
 	return nil
 }
 
+func (m *migration) addNoDataSilence(da dashAlert, rule *alertRule) error {
```

Comment:
If a user later changes the no data state for this alert from `Keep Last State` to something like `Alerting` do we expire the old silence?